etca_asm/base_isa.py

```diff
@@ -223,3 +214,7 @@ def base_jumps(context, inst: str, label: str):
 @base.inst('"nop"')
 def base_nop(context):
     return b"\x8f\x00"  # jump nowhere, never
+
+@base.inst('"halt|hlt"')
```

Comment:
This is the litearl `halt|hlt`, not a regex pattern

Reply:
So I'd change it to `"/halt|hlt/"`?

Reply:
`'/halt|hlt/'`, yes

Reply:
And does that mean that `nop` is incorrect as well?

Reply:
No, that opcode only has the literal syntax `nop`. `"string litearls"` mean match that exact text, `/regex litearls/` mean match this pattern. These are lark expressions, but we only need a single Terminal here so it looks a bit weird. https://github.com/lark-parser/lark .

Also to note: If you change these to regex, the functions will get extra arguments because regex generate terminals in the final result and strings don't. You can also do `'"halt" | "hlt"'`, i.e. one of the terminals. That's probably the best solution tbh.

Reply:
Ah ok. I'll do that then.